go/fn/byteio.go

```diff
@@ -0,0 +1,44 @@
+package fn
+
+import (
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// byteReadWriter wraps kio.ByteReadWriter
+type byteReadWriter struct {
+	kio.ByteReadWriter
+}
+
+func (rw *byteReadWriter) Read() (KubeObjects, *KubeObject, error) {
+	items, err := rw.ByteReadWriter.Read()
+	if err != nil {
+		return nil, nil, err
+	}
+	var result KubeObjects
+	for _, item := range items {
+		obj, err := ParseKubeObject([]byte(item.MustString()))
+		if err != nil {
+			return nil, nil, err
+		}
+		result = append(result, obj)
+	}
+	obj, err := ParseKubeObject([]byte(rw.ByteReadWriter.FunctionConfig.MustString()))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return result, obj, nil
+}
+
+func (rw *byteReadWriter) Write(objs KubeObjects) error {
```

Comment:
I'd suggest to make it write a ResourceList.
We should be able to serialize everything in a resourceList. e.g. Users may want to use `results` field.

Reply:
done! Made this accept a resourceList and moved all the results logic into rw.Write.